c_emulator/riscv_sim.cpp

```diff
@@ -397,14 +421,17 @@ uint64_t load_sail(char *f, bool main_file)
     fprintf(stderr, "Unable to locate tohost symbol; disabling HTIF.\n");
     rv_enable_htif = false;
   } else {
+    rv_htif_tohost += load_offset;
```

Comment:
These need a way to know whether load_elf_at_offset used load_offset or not; at first glance I don't see a way to tell. I suspect Sail needs a lookup_sym_with_(dyn/pie_)offset that is like load_elf_at_offset in conditionally including the PIE (dynamic) offset.